program-runtime/src/timings.rs

```diff
@@ -233,6 +233,18 @@ impl ExecuteDetailsTimings {
             program_timing.count = program_timing.count.saturating_add(1);
         };
     }
+
+    /// sum all programs execution units, include both successful and errored executing
+    pub fn sum_programs_execute_units(&self) -> u64 {
+        self.per_program_timings
+            .iter()
+            .map(|(_key, program_timing)| {
+                program_timing
+                    .total_errored_units
+                    .saturating_add(program_timing.accumulated_units)
+            })
+            .sum()
```

Comment:
As said earlier, I don't like having to sum over all individual program timings twice when the MessageProcessor::process_message() has the execution cu directly available already.

Reply:
Yea, changed to passing sum of units via `&mut Option<&mut u64>`, no more loops ➰